[bug] Runtime panic in v1alpha3 when deleting extra trials

/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]


```
{"level":"info","ts":1592453260.8852422,"logger":"experiment-controller","msg":"DeleteTrials","Experiment":"common1/experiment-20200618015209-kbs4gtfq","deleteCount":1}
E0618 04:07:40.885467       1 runtime.go:69] Observed a panic: "index out of range" (runtime error: index out of range)
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:44
/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:359
/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:283
/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:264
/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:222
/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 292 [running]:
github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1547f40, 0x27efa20)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment.(*ReconcileExperiment).deleteTrials(0xc0008689c0, 0xc000e62000, 0x2827da8, 0x0, 0x0, 0x1, 0xc000d61200, 0x11)
	/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:359 +0x374
github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment.(*ReconcileExperiment).ReconcileTrials(0xc0008689c0, 0xc000e62000, 0x2827da8, 0x0, 0x0, 0xc000401260, 0x0)
	/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:283 +0x29b
github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment.(*ReconcileExperiment).ReconcileExperiment(0xc0008689c0, 0xc000e62000, 0x2827da8, 0x0)
	/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:264 +0x37f
github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment.(*ReconcileExperiment).Reconcile(0xc0008689c0, 0xc000d5b009, 0x7, 0xc000d51680, 0x22, 0x2809e00, 0x0, 0x0, 0x0)
	/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:222 +0x50a
github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0007312c0, 0x0)
	/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x1cc
github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000ab75f0)
	/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000ab75f0, 0x3b9aca00, 0x0, 0x1, 0xc0007254a0)
	/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000ab75f0, 0x3b9aca00, 0xc0007254a0)
	/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x311
```

**What did you expect to happen:**

No panic. We should check the length of the slice.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Kubeflow version:
- Minikube version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

Full log is here, not sure if it is related to the webhook failure.

```
{"level":"info","ts":1592461289.3785846,"logger":"entrypoint","msg":"Config:","experiment-suggestion-name":"default","cert-local-filesystem":true,"webhook-port":8443,"metrics-addr":":8080","inject-security-context":true,"enable-grpc-probe-in-suggestion":false}
{"level":"info","ts":1592461289.5474691,"logger":"entrypoint","msg":"Registering Components."}
{"level":"info","ts":1592461289.547977,"logger":"entrypoint","msg":"Setting up controller"}
{"level":"info","ts":1592461289.5480177,"logger":"experiment-controller","msg":"Using the default suggestion implementation"}
{"level":"info","ts":1592461289.5482223,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"experiment-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1592461289.5486085,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"experiment-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1592461289.5487924,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"experiment-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1592461289.5489528,"logger":"experiment-controller","msg":"Experiment controller created"}
{"level":"info","ts":1592461289.5490866,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"suggestion-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1592461289.54912,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"suggestion-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1592461289.549339,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"suggestion-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1592461289.5495389,"logger":"suggestion-controller","msg":"Suggestion controller created"}
{"level":"info","ts":1592461289.5496511,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"trial-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1592461289.5496879,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"trial-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1592461289.5499399,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"trial-controller","source":"kind source: batch/v1, Kind=Job"}
{"level":"info","ts":1592461289.5501323,"logger":"trial-controller","msg":"Job watch added successfully","CRD Kind":"Job"}
{"level":"info","ts":1592461289.5501583,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"trial-controller","source":"kind source: kubeflow.org/v1, Kind=TFJob"}
{"level":"info","ts":1592461289.5503178,"logger":"trial-controller","msg":"Job watch added successfully","CRD Kind":"TFJob"}
{"level":"info","ts":1592461289.5503542,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"trial-controller","source":"kind source: kubeflow.org/v1, Kind=PyTorchJob"}
{"level":"info","ts":1592461289.55049,"logger":"trial-controller","msg":"Job watch added successfully","CRD Kind":"PyTorchJob"}
{"level":"info","ts":1592461289.550507,"logger":"trial-controller","msg":"Trial  controller created"}
{"level":"info","ts":1592461289.5505133,"logger":"entrypoint","msg":"Setting up webhooks"}
{"level":"info","ts":1592461289.550882,"logger":"entrypoint","msg":"Starting the Cmd."}
{"level":"info","ts":1592461289.9516416,"logger":"kubebuilder.webhook","msg":"installing webhook configuration in cluster"}
{"level":"info","ts":1592461289.9518836,"logger":"kubebuilder.controller","msg":"Starting Controller","controller":"suggestion-controller"}
{"level":"info","ts":1592461289.952075,"logger":"kubebuilder.controller","msg":"Starting Controller","controller":"experiment-controller"}
{"level":"info","ts":1592461289.9520638,"logger":"kubebuilder.controller","msg":"Starting Controller","controller":"trial-controller"}
{"level":"info","ts":1592461290.0521998,"logger":"kubebuilder.controller","msg":"Starting workers","controller":"suggestion-controller","worker count":1}
{"level":"info","ts":1592461290.0522716,"logger":"kubebuilder.controller","msg":"Starting workers","controller":"trial-controller","worker count":1}
{"level":"info","ts":1592461290.052216,"logger":"kubebuilder.controller","msg":"Starting workers","controller":"experiment-controller","worker count":1}
{"level":"info","ts":1592461290.1536314,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"common1/experiment-20200617093759-7dl7fq7q","suggestions":1}
{"level":"info","ts":1592461290.1541686,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"common1/experiment-20200617094159-xh5cmt5j","suggestions":1}
{"level":"info","ts":1592461290.1544588,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"delete/experiment-20200617073138-lkdb2ccc","suggestions":1}
{"level":"info","ts":1592461290.1548774,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"delete/experiment-20200617073449-bmnx2ngn","suggestions":1}
{"level":"info","ts":1592461290.1552343,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"delete/experiment-20200617081502-kdqp62d9","suggestions":3}
{"level":"info","ts":1592461290.1554863,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"common1/experiment-20200617085503-qdvqztc8","suggestions":1}
{"level":"info","ts":1592461290.155779,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"common1/experiment-20200617085738-4lsqmjf4","suggestions":1}
{"level":"info","ts":1592461290.1560652,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"common1/experiment-20200617093910-hq8j7bzk","suggestions":1}
{"level":"info","ts":1592461290.1563156,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"common1/experiment-20200617095132-66n9vdrg","suggestions":1}
{"level":"info","ts":1592461290.1566074,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"delete/experiment-20200617073945-hzsnpssf","suggestions":1}
{"level":"info","ts":1592461291.0514863,"logger":"kubebuilder.admission.cert.writer","msg":"cert directory doesn't exist, creating","directory":"/tmp/cert"}
{"level":"error","ts":1592461291.0873666,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"experiment-controller","request":"delete/experiment-20200618023812-bzqz8zjd","error":"Internal error occurred: failed calling webhook \"mutating.experiment.katib.kubeflow.org\": Post https://katib-controller.kube-system.svc:443/mutate-experiments?timeout=30s: dial tcp 10.254.146.74:443: connect: connection refused","stacktrace":"github.com/kubeflow/katib/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/kubeflow/katib/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\ngithub.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
2020/06/18 06:21:31 http: TLS handshake error from 192.168.64.0:50449: remote error: tls: bad certificate
2020/06/18 06:21:31 http: TLS handshake error from 192.168.65.0:45431: remote error: tls: bad certificate
{"level":"info","ts":1592461292.0881205,"logger":"experiment-controller","msg":"DeleteTrials","Experiment":"common1/experiment-20200618015209-kbs4gtfq","deleteCount":1}
E0618 06:21:32.090798       1 runtime.go:69] Observed a panic: "index out of range" (runtime error: index out of range)
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:44
/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:359
/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:283
/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:264
/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:222
/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 297 [running]:
github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1547f40, 0x27efa20)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment.(*ReconcileExperiment).deleteTrials(0xc00083d860, 0xc000a30780, 0x2827da8, 0x0, 0x0, 0x1, 0xc0006cfb00, 0x11)
	/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:359 +0x374
github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment.(*ReconcileExperiment).ReconcileTrials(0xc00083d860, 0xc000a30780, 0x2827da8, 0x0, 0x0, 0xc0002d1ea0, 0x0)
	/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:283 +0x29b
github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment.(*ReconcileExperiment).ReconcileExperiment(0xc00083d860, 0xc000a30780, 0x2827da8, 0x0)
	/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:264 +0x37f
github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment.(*ReconcileExperiment).Reconcile(0xc00083d860, 0xc0000464e9, 0x7, 0xc0006f8db0, 0x22, 0x2809e00, 0x0, 0x0, 0x0)
	/go/src/github.com/kubeflow/katib/pkg/controller.v1alpha3/experiment/experiment_controller.go:222 +0x50a
github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0007234a0, 0x0)
	/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x1cc
github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000b0c080)
	/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b0c080, 0x3b9aca00, 0x0, 0x1, 0xc00071f620)
	/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000b0c080, 0x3b9aca00, 0xc00071f620)
	/go/src/github.com/kubeflow/katib/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/src/github.com/kubeflow/katib/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x311
```

--- a/pkg/controller.v1alpha3/experiment/experiment_controller.go
+++ b/pkg/controller.v1alpha3/experiment/experiment_controller.go
@@ -18,6 +18,7 @@ package experiment
 
 import (
 	"context"
+	"fmt"
 	"sort"
 
 	"github.com/spf13/viper"
@@ -357,7 +358,7 @@ func (r *ReconcileExperiment) createTrials(instance *experimentsv1alpha3.Experim
 
 func (r *ReconcileExperiment) deleteTrials(instance *experimentsv1alpha3.Experiment,
 	trials []trialsv1alpha3.Trial,
-	deleteCount int32) error {
+	expectedDeletions int32) error {
 	logger := log.WithValues("Experiment", types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()})
 
 	trialSlice := trials
@@ -366,7 +367,12 @@ func (r *ReconcileExperiment) deleteTrials(instance *experimentsv1alpha3.Experim
 			After(trialSlice[j].CreationTimestamp.Time)
 	})
 
-	for i := 0; i < int(deleteCount); i++ {
+	expected := int(expectedDeletions)
+	if len(trialSlice) < expected {
+		return fmt.Errorf("Expected to delete %d trials, but got %d trials",
+			expected, len(trialSlice))
+	}
+	for i := 0; i < expected; i++ {
 		if err := r.Delete(context.TODO(), &trialSlice[i]); err != nil {
 			logger.Error(err, "Trial Delete error")
 			return err

--- a/pkg/controller.v1alpha3/experiment/experiment_controller.go
+++ b/pkg/controller.v1alpha3/experiment/experiment_controller.go
@@ -18,7 +18,6 @@ package experiment
 
 import (
 	"context"
-	"fmt"
 	"sort"
 
 	"github.com/spf13/viper"
@@ -368,9 +367,12 @@ func (r *ReconcileExperiment) deleteTrials(instance *experimentsv1alpha3.Experim
 	})
 
 	expected := int(expectedDeletions)
-	if len(trialSlice) < expected {
-		return fmt.Errorf("Expected to delete %d trials, but got %d trials",
-			expected, len(trialSlice))
+	actual := len(trialSlice)
+	// If the number of trials < expected, we delete all we have.
+	if actual < expected {
+		logger.Info("deleteTrials does not find enough trials, we will delete all trials instead",
+			"expectedDeletions", expected, "trials", actual)
+		expected = actual
 	}
 	for i := 0; i < expected; i++ {
 		if err := r.Delete(context.TODO(), &trialSlice[i]); err != nil {

--- a/pkg/controller.v1beta1/experiment/experiment_controller.go
+++ b/pkg/controller.v1beta1/experiment/experiment_controller.go
@@ -372,7 +372,7 @@ func (r *ReconcileExperiment) createTrials(instance *experimentsv1beta1.Experime
 
 func (r *ReconcileExperiment) deleteTrials(instance *experimentsv1beta1.Experiment,
 	trials []trialsv1beta1.Trial,
-	deleteCount int32) error {
+	expectedDeletions int32) error {
 	logger := log.WithValues("Experiment", types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()})
 
 	trialSlice := trials
@@ -381,7 +381,15 @@ func (r *ReconcileExperiment) deleteTrials(instance *experimentsv1beta1.Experime
 			After(trialSlice[j].CreationTimestamp.Time)
 	})
 
-	for i := 0; i < int(deleteCount); i++ {
+	expected := int(expectedDeletions)
+	actual := len(trialSlice)
+	// If the number of trials < expected, we delete all we have.
+	if actual < expected {
+		logger.Info("deleteTrials does not find enough trials, we will delete all trials instead",
+			"expectedDeletions", expected, "trials", actual)
+		expected = actual
+	}
+	for i := 0; i < expected; i++ {
 		if err := r.Delete(context.TODO(), &trialSlice[i]); err != nil {
 			logger.Error(err, "Trial Delete error")
 			return err

